admin/block-editor-filter-preload-paths.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+/**
+ * Class PLL_Block_Editor_Filter_Preload_Paths
+ */
+class PLL_Block_Editor_Filter_Preload_Paths {
+
+
+	/**
+	 * @var int
+	 */
+	private $arguments_number;
+
+	/**
+	 * @var callable
+	 */
+	private $callback;
+
+	/**
+	 * PLL_Block_Editor_Filter_Preload_Paths constructor.
+	 *
+	 * @since 3.1
+	 *
+	 * @param callable $callback         Function or method to be executed when the filter is triggered.
+	 * @param int      $priority         Priority of execution for the given callback. Default 10.
+	 * @param int      $arguments_number Number of arguments to pass to the callback. Default 1.
+	 */
+	public function __construct( $callback, $priority = 10, $arguments_number = 1 ) {
+		$this->callback = $callback;
+		$this->arguments_number = $arguments_number;
+
+		if ( version_compare( $GLOBALS['wp_version'], '5.8-alpha', '<' ) ) {
```

Comment:
We should check for the existence of `WP_Block_Editor_Context` class instead of WordPress versions (the change of filter names was introduced in an alpha version, but not present in the previous alphas).